symbolic-base/src/ZkFold/Symbolic/Data/UInt.hs

```diff
@@ -73,109 +72,93 @@ import           ZkFold.Symbolic.Data.Input        (SymbolicInput, isValid)
 import           ZkFold.Symbolic.Data.Ord
 import           ZkFold.Symbolic.Interpreter       (Interpreter (..))
 import           ZkFold.Symbolic.MonadCircuit
-
+import Data.Aeson (FromJSON (..), ToJSON (..))
 
 -- TODO (Issue #18): hide this constructor
-newtype UInt (n :: Natural) (r :: RegisterSize) (context :: (Type -> Type) -> Type) = UInt (context (Vector (NumberOfRegisters (BaseField context) n r)))
+newtype UInt (r :: Natural) (n :: Natural) (c :: (Type -> Type) -> Type) = UInt (c (Vector n))
```

Comment:
It might be easy to mess the order of indices, I'd pack at least one of them in a `newtype`